script/demoScripts/demoIntentFactory.ts

```diff
@@ -0,0 +1,138 @@
+import { arbitrum } from 'viem/chains'
+import { defineCommand, runMain } from 'citty'
+import * as Deployments from '../../deployments/arbitrum.staging.json'
+import * as IntentFactory from '../../out/IntentFactory.sol/IntentFactory.json'
+import {
+  Address,
+  Hex,
+  SendTransactionRequest,
+  createPublicClient,
+  createWalletClient,
+  encodeFunctionData,
+  http,
+  keccak256,
+  parseAbi,
+  parseUnits,
+  toHex,
+} from 'viem'
+import { privateKeyToAccount } from 'viem/accounts'
+import { ChainId, getQuote } from '@lifi/sdk'
+
+const INTENT_FACTORY_ADDReSS = Deployments.IntentFactory as Address
+const ABI = IntentFactory.abi
+const ERC20_ABI = parseAbi([
+  'function transfer(address,uint256) external',
+  'function approve(address,uint256) external',
+])
+const DAI = '0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1'
+const USDC = '0xaf88d065e77c8cC2239327C5EDb3A432268e5831'
+const AMOUNT_TO_SWAP = '1000000000000000000'
+
+const main = defineCommand({
+  meta: {
+    name: 'propose-to-safe',
+    description: 'Propose a transaction to a Gnosis Safe',
+  },
+  args: {
+    privateKey: {
+      type: 'string',
+      description: 'Private key of the signer',
+      required: true,
+    },
+  },
+  async run({ args }) {
+    const { privateKey } = args
+    const account = privateKeyToAccount(`0x${privateKey}`)
+    // Read client
+    const publicClient = createPublicClient({
+      chain: arbitrum,
+      transport: http(),
+    })
+
+    // Write client
+    const walletClient = createWalletClient({
+      account,
+      chain: arbitrum,
+      transport: http(),
+    })
+
+    // Initialize the intentfactory
+    const intentFactory = {
+      address: INTENT_FACTORY_ADDReSS,
+      abi: ABI,
+    }
+
+    // Get an initial quote from LIFI
+    let quote = await getQuote({
+      fromAddress: account.address,
+      toAddress: account.address,
+      fromChain: ChainId.ARB,
+      toChain: ChainId.ARB,
+      fromToken: DAI,
+      toToken: USDC,
+      fromAmount: AMOUNT_TO_SWAP,
+    })
+    console.log(quote)
+
+    // Calculate the intent address
+    const intentData = {
```

Comment:
It's meant to be as flexible as possible. This can't be checked anyway since users can send tokens whatever tokens they want to the precomputed address before the intent handler is actually deployed and executed

Reply:
ok